harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -67,3 +178,159 @@ def test_host_poweron_state(api_client, host_state, wait_timeout):
+
+    assert 200 == status_code, f"Failed to list nodes with error: {nodes_info}"
+
+    fails = []
+    for node in nodes_info['data']:
+        node_name = node.get('metadata', {}).get('name', "")
+        if node_name == "":
+            fails.append((node['id'], "node name can't be empty"))
+
+        cpus = node.get('status', {}).get('capacity', {}).get('cpu', 0)
+        if cpus == 0:
+            fails.append((node['id'], "cpu should not be zero"))
+
+        mems = node.get('status', {}).get('capacity', {}).get('memory', "0Ki")
+        mems = "".join(c for c in mems if c.isdigit())
+        mems = mems or "0"
+        if int(mems, 10) == 0:
+            fails.append((node['id'], "memory should not be zero"))
+
+    assert not fails, (
+        "Failed to get node information with errors:\n",
+        "\n".join(f"Node {n}: {r}" for n, r in fails)
+    )
+
+
+@ pytest.mark.hosts
+@ pytest.mark.p0
+def test_maintenance_mode_trigger_vm_migrate(api_client, focal_vm, wait_timeout,
+                                             available_node_names):
+    assert 2 <= len(available_node_names), (
+        f"The cluster only have {len(available_node_names)} available node. \
+            It's not enough for migration test."
+    )
+
+    code, data = api_client.vms.get_status(focal_vm['name'], focal_vm['namespace'])
+    assert 200 == code, (
+        f"Can't get VMI {focal_vm['namespace']}/{focal_vm['name']} with error: {code}, {data}"
+    )
+    src_host = data['status']['nodeName']
+
+    code, data = api_client.hosts.maintenance_mode(src_host, enable=True)
+    assert 200 == code, (
+        f"Failed to enable maintenance mode on node {src_host} with error: {code}, {data}",
+    )
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.hosts.get(src_host)
+        if data.get('metadata', {}) \
+                .get('annotations', {}) \
+                .get('harvesterhci.io/maintain-status', '') == "completed":
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"The maintain-status of node {src_host} can't be completed \
+                with {wait_timeout} timed out\n"
+            f"Got error: {code}, {data}"
+        )
+
+    code, data = api_client.vms.get_status(focal_vm['name'], focal_vm['namespace'])
+    assert 200 == code, (
+        f"Failed to get VM {focal_vm['namespace']}/{focal_vm['name']} with error: {code}, {data}"
+    )
+    assert src_host != data['status']['nodeName'], (
+        f"Failed to migrate VM {focal_vm['namespace']}/{focal_vm['name']} \
+            from {src_host} to another node"
+    )
+    assert 'Running' == data['status']['phase'], (
+        f"Failed to migrate VM {focal_vm['namespace']}/{focal_vm['name']}, \
+            it's not running after migration"
+    )
+
+    # teardown
+    code, data = api_client.hosts.maintenance_mode(src_host, enable=False)
```

Comment:
If migration fails, the host will remain in the maintenance mode, which will lead the subsequent tests failure.

Reply:
I think this should be fine, we will need to check whether the fail is caused by node or VM.

Reply:
Yeah, but we don't want to have the host in maintenance mode if this test fails. It might cause problems for other tests which runs after this test.